### ISSUE TITLE
Content type builder extension installed but not visible in menu

### ISSUE THREAD

**Issue by drieshooghe** [strapi/strapi#2220]
**Informations**
- **Node.js version**:  v11.0.0
- **npm version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.14.4.0
- **Database**: MongoDB
- **Operating system**: Debian 9

**What is the current behavior?**
I just did a fresh install and configuration of Strapi and even though the 'Content Type Builder' extension is displayed as enabled in the Plugins page, the plugin isn't listed in the menu.

![image](https://user-images.githubusercontent.com/25033847/47619086-19c78800-dadb-11e8-93c4-106d3e8e17c5.png)

![image](https://user-images.githubusercontent.com/25033847/47619107-5bf0c980-dadb-11e8-94eb-d92fbed9d675.png)

**Steps to reproduce the problem**
Simply followed [this tutorial](https://medium.com/strapi/how-to-deploy-a-strapi-api-on-ubuntu-16-04-17f8fbbf5c5b)
Where I had to I used debian commands for installing the different applications, but they all seem to be working fine.

**Comment by lauriejim** [comment]
Are you in `production` env ?

**Comment by drieshooghe** [comment]
@lauriejim I am, I launched strapi with pm2 with this command, which gave no errors:
```
NODE_ENV=production pm2 start server.js --name api
```

**Comment by lauriejim** [comment]
Check here https://github.com/strapi/strapi/wiki/Create-files-on-Heroku

**Comment by lauriejim** [comment]
The plugin is not available in production env.
You have to create your content type in dev mode, the deploy your ap in your dev mode.

**Comment by drieshooghe** [comment]
Okay, sure explains a lot. Perhaps something to point out in the tutorial.
Much appreciated!

**Comment by martiro** [comment]
> The plugin is not available in production env.
> You have to create your content type in dev mode, the deploy your ap in your dev mode.

@lauriejim 
I had the same problema as drieshooghe, and now that I read the reason I'd like to go deep: 
where is described the deployment process? I cannot find any docs. What does the deployment involves? The "schema" of the CMS (i meand the entities and their relations) o also the data?
Thanks a lot.

**Comment by lauriejim** [comment]
Here is an article of deployment on Ubuntu server - https://medium.com/strapi/how-to-deploy-a-strapi-api-on-ubuntu-16-04-17f8fbbf5c5b

Actually when you deploy your app, it's just files migration. We never do any action on your data.

**Comment by martiro** [comment]
> Here is an article of deployment on Ubuntu server - https://medium.com/strapi/how-to-deploy-a-strapi-api-on-ubuntu-16-04-17f8fbbf5c5b
> 
> Actually when you deploy your app, it's just files migration. We never do any action on your data.

@lauriejim Thank you v very much!
I did some trial today and now everything is clear, but I found something strange adding new instances for my contents: the fields that where declared as "Text" in developement are editable through WYSWYG    editor, but in production they become simple textareas. How can I fix this problem?
Thanks a lot

**Comment by derrickmehaffy** [comment]
@martiro you need to copy over your `core_store` collection in the database

The WYSWYG setting is saved in the database

**Comment by derrickmehaffy** [comment]
@martiro easier to copy the whole collection but what you need is `plugin_content-manager_schema`

**Comment by Otavioensa** [comment]
Is there a specific reason for not being able to create Content in production? I mean, isn't it the purpose of the framework? Allowing content in production so users can add the content on it and make it available for being consumed.

**Comment by derrickmehaffy** [comment]
@Otavioensa The point of the builder is to get people who are not familiar with backend coding able to build models and connect to the database easier. The purpose of the different environments is to force those people to learn proper deployment strategies. Building models in production requires the server to restart meaning if you make a typo or change something and break the server it won't restart due to that error.

Node requires itself to restart to reflect changes in local files IE the model definition files. The flow would normally be:
1) you make changes to your project locally in a development environment
2) you push those changes to source control (Github, Gitkraken, Whatever)
3) you pull those changes into a staging environment (for best results use a mirror of your production database)
4) test the changes in a "like-prod" type environment (Staging)
5) If everything is fine you schedule downtime/maintenance, backup, pull changes and have proper rollback plans if something fucks up.

This is normal everyday best-practices deployment.

**Comment by derrickmehaffy** [comment]
TLDR: **Never** "test" or "build" in production

**Comment by Otavioensa** [comment]
Hi @derrickmehaffy, thanks for your response! 
But I'm still not figuring how I am supposed to allow the user create content in production. It's not clear enough how I'll be able to create content and make it available for being consumed by a blog in production, for example. 
Could you give me some more explanation how I am supposed to do that? 
Just to give you an insight of my problem: I need to be able to let an user to create such content so the content can be available for being consumed via API. Then I'd have my blog page that would consume such content via API's exposed by my Strapi application. How can I do that in production? How can I let my user create such content? Sorry for being insistent on this question but there is something more that I'm missing or misunderstanding about this whole process 
Thanks

**Comment by derrickmehaffy** [comment]
@Otavioensa you don't block people from adding data to the model in production. You block the creation of models.

Models are the "framework" of your API, the data is what is stored.

**Comment by Otavioensa** [comment]
Ok, now I got the relation between the model and the data stored on it. 
I deployed my app but I'm now running into another problem (which I opened an issue since it's not related to this one)
Thanks @derrickmehaffy

**Comment by stephanschonbeck** [comment]
Is there a way to permanently allow the usage of the content type builder in production mode? 
Why should Strapi be any different than other CMS systems where this is possible?
Seems overkill the need to run a development, staging, production setup to use Strapi, it should work out-of-the-box.

For instance, you don't have to create models locally in development mode when using Directus CMS or WordPress custom post types. This can all be created directly from the admin.

Just found out about this "issue" after setting up Directus directly on my production server. Wish there was some documentation on this before I started.

**Comment by derrickmehaffy** [comment]
@stephanschonbeck because the models are saved in json files, and Strapi is written in node. In order for those to take effect the node server has to restart. In production you do not want auto restarting as this creates a lot of temporary outages (dropping requests), that and the potential to break your production server by improperly editing a model.

Good practices when developing a project, simply put, **Don't Test In Production** thus the environments are a way to force newer developers into a good practice of testing your changes before you just push them.

Also Directus is PHP which has no "server" and is just flat files, nodejs operates completely different.

> Is there a way to permanently allow the usage of the content type builder in production mode?

Short answer is no, and there is no plans to allow it.

**Comment by EricCBonet** [comment]
Would it be possible to remove the blue "Edit the content type" button in the production environment?

It's a bit confusing. 

![image](https://user-images.githubusercontent.com/35337798/71512321-e5de2300-2895-11ea-87aa-642f9b4aee09.png)

**Comment by lauriejim** [comment]
Hello! Yeah good point!
In production, there is only a view of the content types, you can't Edit them.
It will be the same into the Edit Entry page. The button still the same even if the Edit action is not possible.
